data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/configuration/ObjectKeyOptions.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An implementation class of path prefix and file pattern configuration Options
+ */
+public class ObjectKeyOptions {
+    private static final String DEFAULT_OBJECT_NAME_PATTERN = "events-%{yyyy-MM-dd'T'hh-mm-ss}";
+
+    @JsonProperty("path_prefix")
+    private String pathPrefix;
+
+    @JsonProperty("name_pattern")
+    private String namePattern = DEFAULT_OBJECT_NAME_PATTERN;

```

Comment:
As noted above, let's not make this configurable for now.

Reply:
Resolved.